### ISSUE TITLE
clang adds `/usr/include` to header search path while cross compiling with mingw-w64

### ISSUE THREAD

**Issue by godvino** [llvm/llvm-project#59871]
Trying to cross compile to Windows using Clang and mingw-w64 results in Clang adding `/usr/include` to its header search path. This breaks compiling some projects that check for the presence of certain headers like `alloca.h` because it then uses the `alloca.h` from the host system which causes errors.

```console
❯ clang --target=x86_64-w64-mingw32ucrt main.c -o main -v
clang version 15.0.6 (Fedora 15.0.6-3.fc37)
Target: x86_64-w64-windows-gnu
Thread model: posix
InstalledDir: /usr/bin
 "/usr/bin/clang-15" -cc1 -triple x86_64-w64-windows-gnu -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name main.c -mrelocation-model pic -pic-level 2 -mframe-pointer=none -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -mms-bitfields -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -fcoverage-compilation-dir=/var/home/godvino/Documents -resource-dir /usr/lib64/clang/15.0.6 -internal-isystem /usr/lib64/clang/15.0.6/include -internal-isystem /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include -internal-isystem /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/usr/include -internal-isystem /usr/include -fdebug-compilation-dir=/var/home/godvino/Documents -ferror-limit 19 -fno-use-cxa-atexit -fgnuc-version=4.2.1 -exception-model=seh -fcolor-diagnostics -faddrsig -o /tmp/main-50542d.o -x c main.c
clang -cc1 version 15.0.6 based upon LLVM 15.0.6 default target x86_64-redhat-linux-gnu
ignoring nonexistent directory "/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/usr/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib64/clang/15.0.6/include
 /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/include
 /usr/include
End of search list.
 "/usr/bin/x86_64-w64-mingw32ucrt-ld" -m i386pep -Bdynamic -o main.exe /usr/x86_64-w64-mingw32ucrt/sys-root/mingw/lib/crt2.o /usr/lib/gcc/x86_64-w64-mingw32ucrt/12.2.1/crtbegin.o -L/usr/lib/gcc/x86_64-w64-mingw32ucrt/12.2.1 -L/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/lib -L/usr/x86_64-w64-mingw32ucrt/sys-root/mingw/mingw/lib -L/usr/lib /tmp/main-50542d.o -lmingw32 -lgcc -lgcc_eh -lmoldname -lmingwex -lmsvcrt -ladvapi32 -lshell32 -luser32 -lkernel32 -lmingw32 -lgcc -lgcc_eh -lmoldname -lmingwex -lmsvcrt -lkernel32 /usr/lib/gcc/x86_64-w64-mingw32ucrt/12.2.1/crtend.o
```

**Comment by mstorsjo** [comment]
I posted a patch (for discussion) that should fix this issue at https://reviews.llvm.org/D141206.

**Comment by mstorsjo** [comment]
This should be mostly fixed now, in fd15cb935d7aae25ad62bfe06fe9f17cea585978.
